transcom/mymove
# [ MB-14242, MB-16543 ] Update office shipment form to display destination type if dest address is unknown

## [Jira ticket MB-14242](https://dp3.atlassian.net/browse/MB-14242) [Jira ticket MB16543](https://dp3.atlassian.net/browse/MB-16543)

## Summary

Updating the Office shipment edit form to display destination type if the destination address is unknown.

Destination type is displayed for retirees.
When working on this noticed a bug that displays destination type and some retiree verbiage for non-retirees. This bug fix is included in this PR.

## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.

### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/run-storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/about/application-setup/milmove-local-client/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/about/development/testing)

### How to test

1. Access the my move app as a customer
2. Create a move. For the orders type select retirement. (I think for duty location you can select Los Angeles AFB to get you into our default local dev office user GBLOC)
3. Create an HHG shipment. 
4. Select that you do not know your destination address yet.
5. Finalize, sign and submit the move
6. Access the office app as a services counselor. Select the move you just created. 
7. You should see a red warning that the destination type is missing: ![image](https://github.com/transcom/mymove/assets/110134470/902d68e2-656e-43dd-8a56-5d587cf1d40d)
8. Select "Edit Shipment"
9. Go down and select "No" for "Does the customer know their delivery address yet?"
10. The Destination type dropdown should still be visible. 
<img width="623" alt="image" src="https://github.com/transcom/mymove/assets/110134470/04512eec-b98b-42c2-ac61-da8f0a2ec9b9">

11. Select a destination type and save.
12. The red warning should disappear

### Also test non-retirees
1. Access the my move app as a customer
2. Create a move. For the orders type select permanent change of station. (I think for duty location you can select Los Angeles AFB to get you into our default local dev office user GBLOC)
3. Create an HHG shipment. 
4. Select that you do not know your destination address yet.
5. Finalize, sign and submit the move
6. Access the office app as a services counselor. Select the move you just created. 
7. Select "Edit Shipment"
8. You should see no delivery type drop down in the delivery location address: 
![image](https://github.com/transcom/mymove/assets/110134470/da67dc83-4830-4384-a3b8-a3e27d8c3590)

9. Go down and select "No" for "Does the customer know their delivery address yet?"
10. The verbiage should say "We can use the zip of their new duty location:" and no destination type dropdown should be visible.
<img width="491" alt="image" src="https://github.com/transcom/mymove/assets/110134470/e531a863-edf2-4b2a-9e71-37c35dd15c3b">



### Frontend


### Backend


### Database

#### Any new migrations/schema changes:


## Screenshots
